k9mail/src/main/java/com/fsck/k9/activity/compose/AttachmentPresenter.java

```diff
@@ -175,6 +185,27 @@ public void processMessageToForward(MessageViewInfo messageViewInfo) {
         }
     }
 
+    public void processMessageToForwardAsAttachment(MessageViewInfo messageViewInfo) throws IOException, MessagingException {
+        boolean isMissingParts = !allPartsAvailable(messageViewInfo, false);
+        if (isMissingParts) {
+            attachmentMvpView.showMissingAttachmentsPartialMessageForwardWarning();
+        } else {
+            File tempFile = File.createTempFile("pre", ".tmp");
+            tempFile.deleteOnExit();
+            FileOutputStream fileOutputStream = new FileOutputStream(tempFile);
+            messageViewInfo.message.writeTo(fileOutputStream);
```

Comment:
hmm.
maybe someone could help out a little?
i understand that one must use the loader for an Attachment object with a link to the MessageViewInfo so that the AttachmentContentLoader can do it's job but i'm not deep enough into K-9 code to do that myself.